### PR TITLE
Add Api Reference into Directus

### PR DESCRIPTION
**Pull request by Nitwel**
grafik
grafik

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -66,6 +68,7 @@
     "mitt": "^2.1.0",
     "mousetrap": "^1.6.5",
     "nanoid": "^3.1.10",
+    "openapi3-ts": "^1.4.0",
```
Comment (rijkvanzanten) [inline app/package.json:71]: What's this / what does it do?

File: app/src/modules/docs/components/sections.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	},
+	{
+		icon: 'format_list_numbered',
+		name: 'Guides',
+		to: '/docs/guides',
+		default: 'readme',
+	},
+	{
+		icon: 'code',
+		name: 'Api Reference',
```
Comment (rijkvanzanten) [inline app/src/modules/docs/components/sections.ts:60]:
```suggestion
		name: 'API Reference',
```

File: app/package.json (hunk 1 above)
Comment (Nitwel) [reply app/package.json:71]: TS Objects for openapi v3.
Helps typing everything and should not have any balast when building.

**Comment by rijkvanzanten** [comment]
@Nitwel Now that the docs are merged separately in https://github.com/directus/next/pull/436, can we make close this PR? Is this the dynamic API reference?

**Comment by Nitwel** [comment]
Yea, I forgot to change the title. This contains the Api Reference now.

**Comment by Nitwel** [comment]
Why is this now in the `docs module` branch and should I create a new PR for this?

**Comment by rijkvanzanten** [comment]
It was a good poc, but not yet ready to go. The original PR did contain a bunch of other little fixes on the regular docs though that I wanted merged in. The static docs have since been cleaned up and are done, same for the /server/specs/oas endpoint. There's some more work to be done on the front-end part

> should I create a new PR for this?

No, I'll take it from here 👍

**Pull request merged**